From 9451934953bf1a4cc8e88179082785f3de5e127a Mon Sep 17 00:00:00 2001
From: Christoph Hellwig <hch@lst.de>
Date: Tue, 4 Aug 2026 05:50:18 -0700
Subject: [PATCH 10540/15333] block: allow REQ_NOWAIT zone management commands

Commit efae226c2ef1 ("block: handle zone management operations
completions") moved all block layer tracking of zone management
operations to the I/O completion handler.  With that REQ_NOWAIT zone
management operations are just fine, so allow them.

Signed-off-by: Christoph Hellwig <hch@lst.de>
Reviewed-by: Damien Le Moal <dlemoal@kernel.org>
Link: https://patch.msgid.link/20260804125038.740388-3-hch@lst.de
Signed-off-by: Jens Axboe <axboe@kernel.dk>
---
 block/blk-zoned.c | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/block/blk-zoned.c b/block/blk-zoned.c
index 034af9dfb5f9..3b7a5f2bdf98 100644
--- a/block/blk-zoned.c
+++ b/block/blk-zoned.c
@@ -1586,15 +1586,6 @@ static bool blk_zone_wplug_handle_zone_mgmt(struct bio *bio)
 		return true;
 	}
 
-	/*
-	 * No-wait zone management BIOs do not make much sense as the callers
-	 * issue these as blocking operations in most cases. To avoid issues
-	 * with the BIO execution potentially failing with BLK_STS_AGAIN, warn
-	 * about REQ_NOWAIT being set and ignore that flag.
-	 */
-	if (WARN_ON_ONCE(bio->bi_opf & REQ_NOWAIT))
-		bio->bi_opf &= ~REQ_NOWAIT;
-
 	return false;
 }
 
-- 
2.55.0

